metronome/checkpointing/interpreter/src/io/iohk/metronome/checkpointing/interpreter/messages/InterpreterMessage.scala

```diff
@@ -167,9 +180,135 @@ object InterpreterMessage extends RPCMessageCompanion {
       with FromService
       with NoResponse
 
+  /** Interpreter <=> Chain */
+
+  /** The interpreter requests the chain for block at target height in order to construct
+    * the service block
+    */
+  case class CheckpointBlockRequest(
+      requestId: RequestId,
+      blockNumber: BigInt
+  ) extends InterpreterMessage
+      with Request
+      with FromInterpreter
```

Comment:
If you read the definition of the `InterpreterMessage`, it says:
```scala
/** Messages exchanged between the Checkpointing Service
  * and the local Checkpointing Interpreter.
  */
sealed trait InterpreterMessage { self: RPCMessage => }
```

You cannot extend it to also contain messages that go between the ETC chain and the interpreter, the two are completely separate things.